File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithRecursiveField/expected/client/PersonProjection.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithRecursiveField.expected.client;

import com.netflix.graphql.dgs.client.codegen.BaseSubProjectionNode;

public class PersonProjection<PARENT extends BaseSubProjectionNode<?, ?>, ROOT extends BaseSubProjectionNode<?, ?>> extends BaseSubProjectionNode<PARENT, ROOT> {
  public PersonProjection(PARENT parent, ROOT root) {
    super(parent, root, java.util.Optional.of("Person"));
  }

  public PersonProjection<PARENT, ROOT> __typename() {
    getFields().put("__typename", null);
    return this;
  }

  public PersonProjection<PersonProjection<PARENT, ROOT>, ROOT> friends() {
     PersonProjection<PersonProjection<PARENT, ROOT>, ROOT> projection = new PersonProjection<>(this, getRoot());
     getFields().put("friends", projection);
     return projection;
  }

  public PersonProjection<PARENT, ROOT> firstname() {
    getFields().put("firstname", null);
    return this;
  }

  public PersonProjection<PARENT, ROOT> lastname() {
    getFields().put("lastname", null);
    return this;
  }
}

File: graphql-dgs-codegen-core/src/test/resources/kotlin2/dataClassWithRecursiveField/expected/client/PersonProjection.kt
package kotlin2.dataClassWithRecursiveField.expected.client

import com.netflix.graphql.dgs.client.codegen.GraphQLProjection

public class PersonProjection : GraphQLProjection() {
  public val firstname: PersonProjection
    get() {
      field("firstname")
      return this
    }

  public val lastname: PersonProjection
    get() {
      field("lastname")
      return this
    }

  public fun friends(_projection: PersonProjection.() -> PersonProjection): PersonProjection {
    project("friends", PersonProjection(), _projection)
    return this
  }
}
